__tests__/runtime.ts

```diff
@@ -0,0 +1,83 @@
+import {test, expect} from '@jest/globals'
+
+import transform from '../helpers/transform'
+
+const code = `const {foo, ...rest} = obj`
+const targets = 'IE 11' // ensures that a helper is required to transpile the rest operator
+
+test('imports external runtime helpers by default', () => {
+  const options = {targets}
+  const development = transform({code, options, env: 'development'})
+  const production = transform({code, options, env: 'production'})
+  const esm = transform({code, options, env: 'esm'})
+  const cjs = transform({code, options, env: 'cjs'})
+
+  expect(development).toMatchInlineSnapshot(`
+    import _objectWithoutPropertiesLoose from '@babel/runtime/helpers/objectWithoutPropertiesLoose'
+    var _obj = obj,
+      foo = _obj.foo,
+      rest = _objectWithoutPropertiesLoose(_obj, ['foo'])
+  `)
+
+  expect(production).toBe(development)
+  expect(esm).toBe(development)
+
+  expect(cjs).toMatchInlineSnapshot(`
+    'use strict'
+    var _interopRequireDefault = require('@babel/runtime/helpers/interopRequireDefault').default
+    var _objectWithoutPropertiesLoose2 = _interopRequireDefault(
+      require('@babel/runtime/helpers/objectWithoutPropertiesLoose')
+    )
+    var _obj = obj,
+      foo = _obj.foo,
+      rest = (0, _objectWithoutPropertiesLoose2.default)(_obj, ['foo'])
+  `)
+})
+
+test('injects runtime helpers when `runtime` is disabled', () => {
+  const options = {targets, runtime: false}
+  const development = transform({code, options, env: 'development'})
+  const production = transform({code, options, env: 'production'})
+  const esm = transform({code, options, env: 'esm'})
+  const cjs = transform({code, options, env: 'cjs'})
```

Comment:
Instead of defining variables for transforms, let's define a single variable for the snapshot being compared against. This would help structure the expect assertions below in the same way as they are being used in some of the other test files, by keeping the transform params close to the assertion.

Reply:
Alternatively, we could defined a meta-assertion that can accept a list of transforms and compare them against a single snapshot. This way we could avoid thinking about variables entirely

Reply:
>Instead of defining variables for transforms, let's define a single variable for the snapshot being compared against.

Could you show a snippet of what you mean?

Reply:
I was thinking along the lines of
```
[
  transform({code, options, env: 'development'}),
  transform({code, options, env: 'production'}),
  transform({code, options, env: 'esm'})
].forEach(snapshot => {
  expect(snapshot).toBe(`
    function _objectWithoutPropertiesLoose(source, excluded) {
      if (source == null) return {}
      var target = {}
      var sourceKeys = Object.keys(source)
      var key, i
      for (i = 0; i < sourceKeys.length; i++) {
        key = sourceKeys[i]
        if (excluded.indexOf(key) >= 0) continue
        target[key] = source[key]
      }
      return target
    }
    var _obj = obj,
      foo = _obj.foo,
      rest = _objectWithoutPropertiesLoose(_obj, ['foo'])
  `)
})
```
when we assert that multiple transforms map to a single snapshot. This avoids having to trace which snapshot variable is the source of truth, and their relations

Reply:
I think maybe it wasn't clear in the PR body that the idea of inline snapshots is that you don't write the snapshot, Jest does. Unlike regular snapshots, it writes them inline in the assertion call instead of into a separate file.

I don't think it makes sense to consider solutions that can't auto-generate and update the output. (Each inline snapshot assertion can only be called once, by design.)

Reply:
Gotcha, wasn't aware of the limitations of inline snapshots. Is it still possible to compare the transform outputs against some source of truth without storing them as variables? Eg `expect(transform(...)).toBe(development)`

Reply:
I think I'm a bit confused by the line of questioning in this thread. Is `development` something like `const development = transform(...)`? If so, then you can definitely make that assertion because `transform` is a regular function that returns a string, so you're comparing two strings together. But then you've stored the transform output as a variable, which it seems like you're trying to avoid, although I don't necessarily understand why. Did you mean to write `.toMatchInlineSnapshot`? If so, then I don't understand what `development` would be.

Maybe it would be helpful to back up; could you explain broadly what you dislike about the current code, without getting into potential solutions?

That said, it seems like the consensus is otherwise to go with the https://github.com/kensho-technologies/babel-preset/pull/118 approach so maybe this is moot.